tests/appsec/iast/utils.py

```diff
@@ -41,7 +41,12 @@ def assert_iast_vulnerability(
         vulns = [v for v in vulns if v["type"] == vulnerability_type]
         assert vulns, f"No vulnerability of type {vulnerability_type}"
     if expected_location:
-        vulns = [v for v in vulns if v.get("location", {}).get("path", "") == expected_location]
+        vulns = [
+            v
+            for v in vulns
+            if (context.library < "java@1.47.0" and v.get("location", {}).get("path", "") == expected_location)
```

Comment:
can you extract this logic to a line above, with something like `old_version` like the other case? embedding the whole thing within the for comprehension gets unwieldy.